src/dvc_data/index.py

```diff
@@ -175,6 +176,7 @@ def build(index, path, fs, **kwargs):
             obj = None
             hash_info = None
 
+        entry.odb = odb
```

Comment:
So entry.cache is still your repo's local cache, while `entry.odb` is previosly ignored staging odb.

Reply:
E.g.

push: transfer from `entry.cache` to `entry.remote`
fetch: transfer from `entry.remote` to `entry.cache`
commit: transfer from `entry.odb` (e.g. `staging`) to `entry.cache`
commit to-remote: transfer from `entry.odb` to `entry.remote`
etc